.github/workflows/protobuf.yaml

```diff
@@ -0,0 +1,47 @@
+name: protobuf
+permissions: read-all
+
+on:
+  pull_request:
+    branches:
+    paths:
+      - '.github/workflows/protobuf.yaml'
+      - 'protobuf/**'
+  push:
+    branches: [ master ]
+    paths:
+      - '.github/workflows/protobuf.yaml'
+      - 'protobuf/**'
+  schedule:
+    - cron: '0 0 * * 0' # weekly
+
+jobs:
+  build:
+    defaults:
+      run:
+        working-directory: protobuf
+    strategy:
+      matrix:
+        sdk: [stable, dev]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        include:
+          - sdk: stable
+            run-tests: true
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9
+      - uses: dart-lang/setup-dart@e51d8e571e22473a2ddebf0ef8a2123f0ab2c02c
+        with:
+          sdk: ${{matrix.sdk}}
+
+      - run: dart pub get
+
+      - run: dart analyze --fatal-infos
+
+      - run: dart format --output=none --set-exit-if-changed .
+        if: ${{matrix.run-tests}}
+
+      - run: dart test
+        if: ${{matrix.run-tests}}
```

Comment:
If one of these steps fail the rest won't run, right?

That's not ideal because then a formatting issue blocks test results. With multiple packages in the same repo that need preparation steps before testing, it's not uncommon to rely on CI for the test results.

So we should make these steps independent if possible and try to collect as many results as possible from each CI run.